src/main/java/org/moe/idea/sdk/MOESdkType.java

```diff
@@ -178,48 +140,23 @@ public boolean isRootTypeApplicable(@NotNull OrderRootType type) {
                 type == AnnotationOrderRootType.getInstance();
     }
 
-    @NotNull
-    private static SdkModificator getAndInitialiseSdkModificator(
-            @NotNull Sdk sdk,
-            @Nullable Sdk jdk
-    ) {
-        final SdkModificator sdkModificator = sdk.getSdkModificator();
-        final MOESdkAdditionalData data = new MOESdkAdditionalData(sdk, jdk);
-        sdkModificator.setSdkAdditionalData(data);
-        if (jdk != null) {
-            sdkModificator.setVersionString(jdk.getVersionString());
-        }
-        return sdkModificator;
-    }
-
-    public static Sdk getJDK() {
-        Sdk sdk = null;
-
-        for(Sdk jdk: ProjectJdkTable.getInstance().getAllJdks()) {
-            if(jdk != null && jdk.getSdkType() instanceof JavaSdk) {
-
-                if(sdk == null) {
-                    sdk = jdk;
-                }
-                else {
-                    JavaSdkVersion version = ((JavaSdk) jdk.getSdkType()).getVersion(jdk);
-                    JavaSdkVersion desiredVersion = ((JavaSdk)sdk.getSdkType()).getVersion(sdk);
-
-                    if(version != null && desiredVersion != null && version.isAtLeast(desiredVersion)) {
-                        sdk = jdk;
-                    }
-                }
-            }
-        }
-
-        return sdk;
-    }
-
+    /**
+     * Get or create a new MOE SDK.
+     *
+     * If the given module is not configured with a MOE SDK, then this function will
+     * run the Gradle task to obtain the SDK path and create a new SDK. Otherwise the existing
+     * SDK will be returned.
+     */
     public static Sdk getMOESdk(Module module) {
         if (module == null) {
             return null;
         }
 
+        Sdk currentSdk = ModuleRootManager.getInstance(module).getSdk();
+        if (currentSdk != null && isMOESdk(currentSdk) && currentSdk.getHomePath() != null) {
```

Comment:
Maybe we should check here, whether the moeRootPath match? Otherwise I could imagine that on a moe update, still the old MoeSDKType would be selected here

Reply:
Ah that's a good point.